documentation/src/main/asciidoc/migration/index.asciidoc

```diff
@@ -0,0 +1,3351 @@
+
+It would look like this in Hibernate Search 6:
+
+[source, JAVA]
+----
+SearchSession session = Search.session( entityManager );
+List<List<?>> hits = searchSession.search( Book.class )
+        .select( f -> f.composite(
+        		f.field( "isbn" ),
+        		f.field( "title" )
+        ) )
+        .where( f -> f.matchAll() )
+        .fetchHits( 20 );
+----
+
+Alternatively, the composite projection can be made more type-safe:
+
+[source, JAVA]
+----
+SearchSession session = Search.session( entityManager );
+List<Map.Entry<ISBN, String>> hits = searchSession.search( Book.class )
+        .select( f -> f.composite(
+        		Map::entry,
+        		f.field( "isbn", ISBN.class ),
+        		f.field( "title", String.class )
+        ) )
+        .where( f -> f.matchAll() )
+        .fetchHits( 20 );
+----
+
+[[projections-reference]]
+===== `ProjectionConstants`/`ElasticsearchProjectionConstants` migration reference
+
+The code below makes some assumptions for Hibernate Search 6: the projection is being built in a lambda expression,
+e.g. `Search.session( entityManager ).search( Book.class ).select( f -> ... )`.
+
+[cols="l,l,a",options="header"]
+|====
+|Hibernate Search 5 `ProjectionConstants` or `ElasticsearchProjectionConstants`|Hibernate Search 6|Documentation
+|THIS
+|f.entity()
+|link:{hibernateSearchDocUrl}#search-dsl-projection-entity[`entity`]
+
+|DOCUMENT
+|f.extension(LuceneExtension.get())
+        .document()
+|link:{hibernateSearchDocUrl}#search-dsl-projection-extensions-lucene-document[`document`]
+
+|SCORE
+|f.score()
+|link:{hibernateSearchDocUrl}#search-dsl-projection-score[`score`]
+
+|ID
+|f.composite(EntityReference::id,
+        f.entityReference())
+|link:{hibernateSearchDocUrl}#search-dsl-projection-reference[`entityReference`],
+link:{hibernateSearchDocUrl}#search-dsl-projection-composite[`composite`]
+
+Alternatively, `f.entityReference()` (without the wrapping in the "composite" projection)
+will return an instance of `EntityReference`,
+which includes both the type and ID of the entity.
+
+|DOCUMENT_ID
+|f.composite(DocumentReference::id,
+        f.documentReference())
+|link:{hibernateSearchDocUrl}#search-dsl-projection-documentReference[`documentReference`],
+link:{hibernateSearchDocUrl}#search-dsl-projection-composite[`composite`]
+
+This no longer returns the internal Lucene document ID
+(which can change from one query execution to the next),
+but instead returns the Hibernate Search document ID,
+i.e. the String version of the property annotated with `@Id` or `@DocumentId`.
+Note that you no longer need the internal Lucene document ID
+to get an <<searching-fulltextquery-explain,explanation of the score>>.
+
+Alternatively, `f.documentReference()` (without the wrapping in the "composite" projection)
+will return an instance of `DocumentReference`,
+which includes both the type and ID of the document.
+
+|EXPLANATION
+|// Lucene
+f.extension(LuceneExtension.get())
+        .explanation()
+// Elasticsearch
```

Comment:
this line is not visible from my rendered page

Reply:
Fixed